cloudferrylib/utils/bases.py

```diff
@@ -0,0 +1,84 @@
+# Copyright 2016 Mirantis Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+def sorted_field_names(obj):
+    """
+    Returns alphabetically sorted list of public object field names (i.e. their
+    names don't start with '_')
+    """
+
+    return sorted(
+        f for f in dir(obj)
+        if not f.startswith('_') and not hasattr(getattr(obj, f), '__call__'))
+
+
+def compute_hash(obj):
+    """
+    Hash function that is able to compute hashes for lists and dictionaries.
+    """
+
+    if isinstance(obj, dict):
+        return hash_iterable(sorted(obj.items()))
+    elif hasattr(obj, '__iter__'):
+        return hash_iterable(obj)
+    else:
+        return hash(obj)
+
+
+def hash_iterable(iterable):
+    """
+    Compute hash for some iterable value
+    """
+    value = hash(iterable.__class__)
+    for item in iterable:
+        value = ((value * 1000003) & 0xFFFFFFFFFFFFFFFF) ^ compute_hash(item)
```

Comment:
I'd suggest `sys.maxint` as mask, more chances to survive on different architectures